src/cli/cmd/iexec-app.js

```diff
@@ -1038,27 +1040,23 @@ run
 
       debug('requestorder', requestorder);
 
-      const totalCost = new BN(requestorder.appmaxprice)
-        .add(new BN(requestorder.datasetmaxprice))
-        .add(new BN(requestorder.workerpoolmaxprice))
-        .mul(new BN(requestorder.volume));
-
-      const { stake } = await checkBalance(chain.contracts, requester);
-      if (totalCost.gt(stake)) {
-        throw Error(
-          `Not enough RLC on your account (${formatRLC(
-            totalCost,
-          )} RLC required). Run "iexec account deposit" to topup your account.`,
-        );
-      }
+      const { total: totalCost, sponsored } = await estimateMatchOrders(
+        chain.contracts,
+        chain.voucherHub,
+        apporder,
+        datasetorder,
+        workerpoolorder,
+        requestorder,
+        opts.useVoucher,
+      );
 
       spinner.stop();
 
       if (!opts.force) {
         await prompt.custom(
-          `Do you want to spend ${formatRLC(
-            totalCost,
-          )} RLC to execute the following request: ${pretty({
+          `Do you want to spend ${formatRLC(totalCost)} nRLC ${
+            sponsored > 0 ? `(${sponsored} nRLC sponsored by the voucher)` : ''
+          } to execute the following request: ${pretty({
```

Comment:
Before `totalCost` was expressed in RLC, now it is nRLC?
Or should it remain as RLC in this sentence?

```suggestion
          `Do you want to spend ${formatRLC(totalCost)} RLC ${
            sponsored > 0 ? `(${sponsored} RLC sponsored by the voucher)` : ''
          } to execute the following request: ${pretty({
```

Reply:
before, totalCost was calculated manually (I removed that code and replaced it with estimateMatchOrders), and now this function calculates both totalCost and sponsored in nRLC